### DISCUSSION TITLE
Detail Page of deleted item still available(without data) after clicking back button in the browser

### DISCUSSION THREAD

**Discussion by alex96dev** [neuefische/questions#816] [Web]
By sending a http DELETE Request we can successfully delete an item with the proper id in our mongoDB database. The issue is that we are still able to go back to the Detail Page with the deleted item-id in the URL by clicking the back button in the browser or by manually paste it in the address bar. We then receive the rendered Detail Page without any data in the form because the item is already deleted in the mongoDB as I mentioned before.

(User Story: https://github.com/users/alex96dev/projects/1?pane=issue&itemId=55025292)

**Comment by NikRoe** [comment]
Hi Alex,

you can consider implementing a `useEffect` in your dynamic route that uses `router.replace` (similar to what you already have) and redirects to a custom 404 page, whenever a user tries to go to a route that does not exist they will be forwarded to your 404 page.

The useEffect could look similar to this, but please adapt it to your application's needs:
```js
useEffect(() => {
    if (!activities && !isLoading && !error) {
      // Replace this condition with your logic to check if the route exists
      router.replace("/404"); // Redirect to the custom 404 page
    }
  }, [activities, router, isLoading, error]);
```

In case you are interested in a custom error page, take a look at [this article from Next.js.](https://nextjs.org/docs/pages/building-your-application/routing/custom-error).

I hope this already helps you!

Happy coding ✌️
